src/main/java/dk/kb/util/webservice/OpenApiResource.java

```diff
@@ -0,0 +1,212 @@
+        }
+    }
+
+    /**
+     * Deliver the OpenAPI specification with substituted configuration values as a JSON file.
+     */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/{path}.json")
+    public Response getJsonSpec(@PathParam("path") String path){
+        try {
+            return createJson(path);
+        } catch (Exception e) {
+            throw handleException(e);
+        }
+    }
+
+    static Response createJson(String path) {
+        try {
+            path = new File(path).getName();
+            String inputYaml = Resolver.readFileFromClasspath(path + ".yaml");
+
+            if (inputYaml == null){
+                // We want to see if people are trying to hack their way in by trying different paths
+                log.warn("No OpenAPI specification with path '{}' was found.", path);
+                throw new FileNotFoundException("No OpenAPI specification with path '" + path + ".yaml' was found.");
+            }
+
+            String correctString = OpenApiResource.replaceConfigPlaceholders(inputYaml);
+
+            String jsonString = getJsonString(correctString);
+
+            Response.ResponseBuilder builder = Response.ok(jsonString).header("Content-Disposition", "inline; filename=" + path + ".json");
+            return builder.build();
+        } catch (IOException | RuntimeException e){
+            log.warn("Unable to dynamically enhance the YAML OpenAPI specification with path '{}'", path, e);
+            throw new NotFoundServiceException(
+                    "Unable to dynamically enhance the YAML OpenAPI specification with path '" + path + ".yaml'");
+        }
+    }
+
+    /*@GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/openapi.json")
+    public Response getJsonShorthand(){
+        try {
+            return createJson("ds-discover-openapi_v1");
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }*/
+
+    /**
+     * Replace placeholders in the original OpenAPI YAML specification. These placeholders have the format ${config.yamlpath},
+     * where the value inside the '{}' and after 'config.' is treated as a YAML path which is used to find the
+     * replacement value in the backing configuration files.
+     * @param originalApiSpec the content of the original YAML specification
+     * @return an updated YAML string, where config placeholders have been replaced.
+     */
+    private static String replaceConfigPlaceholders(String originalApiSpec) {
+        return CONFIG_PROCESSOR.apply(originalApiSpec);
+    }
+
+    /**
+     * Resolve the value for the given YAML path in the configuration files for the project.
+     * @param yPath to extract value from.
+     * @return the value at the given path in the configuration files.
+     */
+    private static String getReplacementForMatch(String yPath) {
+        return config.get(yPath).toString();
```

Comment:
Last thing, promise! Please add a check to see if `config` has been set here and throw an exception (`IllegalStateException` maybe?) stating that `config` must be initialized before use. No need for a full explanation on how to do so: "See OpenApiResource Javadoc for details` is fine.